DirectProgramming/DPC++FPGA/Tutorials/GettingStarted/fpga_template/src/CMakeLists.txt

```
@@ -0,0 +1,104 @@
+###############################################################################
+### Customize these build variables
+###############################################################################
+set(SOURCE_FILE fpga_template.cpp)
+set(TARGET_NAME fpga_template)
+
+# When targeting a board, the `FPGA_BOARD` will be targeted. Options are:
+#   intel_s10sx_pac:pac_s10
+#   intel_s10sx_pac:pac_s10_usm
+#   intel_a10gx_pac:pac_a10
+set(FPGA_BOARD "intel_s10sx_pac:pac_s10_usm")
+
+set(USER_HARDWARE_FLAGS "")
+# use cmake -DUSER_HARDWARE_FLAGS=<flags> to set extra flags for FPGA backend
+# compilation
+
+set(USER_FLAGS "")
+# use cmake -DUSER_FLAGS=<flags> to set extra flags for general compilation
+
+###############################################################################
+### no changes after here
+###############################################################################
+
+# define the different build targets the generated build script will support.
+set(EMULATOR_TARGET ${TARGET_NAME}.fpga_emu)
+
+# Full-system
+set(BOARD_REPORT_TARGET ${TARGET_NAME}.fpga_report)
+set(BOARD_SIMULATOR_TARGET ${TARGET_NAME}.fpga_sim)
+set(BOARD_TARGET ${TARGET_NAME}.fpga)
+
+# FPGA board selection
+if(DEFINED FPGA_BOARD)
```

```suggestion
if(DEFINED FPGA_DEVICE)
```
With your cmake structure, FPGA_DEVICE will always be defined, so no need for a test?

---

leave in case user overrides